internal/p2p/router.go

```diff
@@ -642,6 +634,7 @@ func (r *Router) connectPeer(ctx context.Context, address NodeAddress) {
 
 	if err := r.runWithPeerMutex(func() error { return r.peerManager.Dialed(address) }); err != nil {
 		r.logger.Error("failed to dial peer", "op", "outgoing/dialing", "peer", address.NodeID, "err", err)
+		r.peerManager.dialWaker.Wake()
```

Comment:
Why don't we call `DialFailed` here?

Reply:
because the dialing succeeded we just throw out the connection because this is your race bug.

Reply:
I think we should wake the dial sleeper from in the `Dialed` method for the errors where we expect to need to start a new dial, namely:

https://github.com/tendermint/tendermint/blob/83526cacbcd998b33d2669088a688c7e492da8ff/internal/p2p/peermanager.go#L672

and 

https://github.com/tendermint/tendermint/blob/83526cacbcd998b33d2669088a688c7e492da8ff/internal/p2p/peermanager.go#L675